sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslAggregationNullableTest.java

```diff
@@ -0,0 +1,156 @@
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Integration tests for aggregation nullable columns. */
+public class BeamSqlDslAggregationNullableTest {
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  private PCollection<Row> boundedInput;
+
+  @Before
+  public void setUp() {
+    Schema schema =
+        Schema.builder()
+            .addNullableField("f_int1", Schema.FieldType.INT32)
+            .addNullableField("f_int2", Schema.FieldType.INT32)
+            .addInt32Field("f_int3")
+            .build();
+
+    List<Row> rows =
+        TestUtils.RowsBuilder.of(schema)
+            .addRows(1, 5, 1)
+            .addRows(null, 1, 1)
+            .addRows(2, 1, 1)
+            .addRows(null, 1, 1)
+            .addRows(null, null, 1)
+            .addRows(null, null, 1)
+            .addRows(3, 2, 1)
+            .getRows();
+
+    boundedInput = PBegin.in(pipeline).apply(Create.of(rows).withCoder(schema.getRowCoder()));
+  }
+
+  @Test
+  public void testCount() {
+    String sql = "SELECT COUNT(f_int1) FROM PCOLLECTION GROUP BY f_int3";
+
+    PAssert.that(boundedInput.apply(SqlTransform.query(sql))).satisfies(matchesScalar(3L));
+
+    pipeline.run();
+  }
+
+  @Test
+  public void testCountStar() {
+    String sql = "SELECT COUNT(*) FROM PCOLLECTION GROUP BY f_int3";
+
+    PAssert.that(boundedInput.apply(SqlTransform.query(sql))).satisfies(matchesScalar(7L));
+
+    pipeline.run();
+  }
+
+  @Test
+  public void testCountThroughSum() {
+    String sql =
+        "SELECT SUM(CASE f_int1 IS NOT NULL WHEN TRUE THEN 1 ELSE 0 END) "
+            + "FROM PCOLLECTION GROUP BY f_int3";
+
+    PAssert.that(boundedInput.apply(SqlTransform.query(sql))).satisfies(matchesScalar(3));
+
+    pipeline.run();
+  }
+
+  @Test
+  public void testCountNulls() {
+    String sql =
+        "SELECT SUM(CASE f_int1 IS NULL WHEN TRUE THEN 1 ELSE 0 END) "
+            + "FROM PCOLLECTION GROUP BY f_int3";
+
+    PAssert.that(boundedInput.apply(SqlTransform.query(sql))).satisfies(matchesScalar(4));
+
+    pipeline.run();
+  }
+
+  @Test
+  public void testSum() {
+    String sql = "SELECT SUM(f_int1) FROM PCOLLECTION GROUP BY f_int3";
+
+    PAssert.that(boundedInput.apply(SqlTransform.query(sql))).satisfies(matchesScalar(6));
+
+    pipeline.run();
+  }
+
+  @Test
+  public void testAvg() {
+    String sql = "SELECT AVG(f_int1) FROM PCOLLECTION GROUP BY f_int3";
+
+    PAssert.that(boundedInput.apply(SqlTransform.query(sql))).satisfies(matchesScalar(2));
+
+    pipeline.run();
+  }
+
+  @Ignore
+  // FIXME java.lang.IllegalArgumentException: Field EXPR$0 is not nullable
```

Comment:
Can you file a Jira with more details for this one?

Reply:
I can file the Jira. I noticed before it related to all other aggregation functions (like `avg`, `sum`)

Reply:
Awesome, please do.

Reply:
Created a ticket: https://issues.apache.org/jira/browse/BEAM-5056